src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -2435,7 +2435,8 @@ else if (!useTnir) {
 
         }
 
-        assert result.equals(Result.SUCCESS) == true;
+        boolean equalSuccess = result.equals(Result.SUCCESS);
```

Comment:
I agree about rule & purpose.  I feel that this might not be good scenario for this rule because you are not performing any operation. Performing any operation in assert is **bad coding**

**Think about this in different way:**
What will happen if result is not Result.SUCCESS if assertions are not enabled.
Are we handling this scenario / condition in somewhere (may be in our exception flow)? Should we want to handle it here? 

if not then what value addition we are doing here?